python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py

```diff
@@ -779,6 +779,146 @@ def evaluate_with_onnx(self, data,
+            invalidInputError(validation_data is not None,
+                              "When call predict_interval for the first time, you must pass in "
+                              "validation data to calculate data noise.")
+            invalidOperationError("mse" in self.metrics,
+                                  "You must add 'mse' in metrics when initialize Forecaster.")
+            output = self.evaluate(data=validation_data, multioutput='uniform_average')
+            index = self.metrics.index("mse")
+            self.data_noise = output[index]
+
+        # step2: data preprocess
+        if isinstance(data, TSDataset):
+            _rolled = data.numpy_x is None
+            data = data.to_torch_data_loader(batch_size=batch_size,
+                                             roll=_rolled,
+                                             lookback=self.data_config['past_seq_len'],
+                                             horizon=self.data_config['future_seq_len'],
+                                             feature_col=data.roll_feature,
+                                             target_col=data.roll_target,
+                                             shuffle=False)
+        # data transform
+        is_local_data = isinstance(data, (np.ndarray, DataLoader))
+        if is_local_data and self.distributed:
+            if isinstance(data, DataLoader):
+                invalidInputError(False,
+                                  "We will be support input dataloader later.")
+            data = np_to_xshard(data)
+        if not is_local_data and not self.distributed:
+            data = xshard_to_np(data, mode="predict")
+
+        def calculate(data, model):
+            if self.distributed:
+                yhat = model.predict(data, batch_size=batch_size)
+                expand_dim = []
+                if self.data_config["future_seq_len"] == 1:
+                    expand_dim.append(1)
+                if self.data_config["output_feature_num"] == 1:
+                    expand_dim.append(2)
+                if is_local_data:
+                    yhat = xshard_to_np(yhat, mode="yhat", expand_dim=expand_dim)
+                else:
+                    yhat = yhat.transform_shard(xshard_expand_dim, expand_dim)
+                return yhat
+            else:
+                if not self.fitted:
+                    invalidInputError(False,
+                                      "You must call fit or restore first before calling predict!")
+                self.internal.eval()
+                yhat = _pytorch_fashion_inference(model=model,
+                                                  input_data=data,
+                                                  batch_size=batch_size)
+                if not is_local_data:
+                    yhat = np_to_xshard(yhat, prefix="prediction")
+                return yhat
+
+        # step3: calculate y_hat
+        y_hat = calculate(data, self.internal)
```

Comment:
should we use the mean y_hat as the final result?

Reply:
yeah, will use mean y_hat